genie-web/src/main/java/com/netflix/genie/web/services/impl/S3AttachmentService.java

```diff
@@ -0,0 +1,86 @@
+/*
+ *
+ *  Copyright 2020 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.genie.web.services.impl;
+
+import com.netflix.genie.common.exceptions.GenieException;
+import com.netflix.genie.web.exceptions.checked.SaveAttachmentException;
+import com.netflix.genie.web.services.AttachmentService;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.core.io.Resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Implementation of the AttachmentService interface which saves attachments to AWS S3.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+@Slf4j
+public class S3AttachmentService implements AttachmentService {
+
+    @Override
+    public Set<URI> saveAttachments(final String jobId, final Set<Resource> attachments) throws SaveAttachmentException {
+        // Check for limits (individual attachments size and total size)
```

Comment:
The limits should already have been handled by spring before it's even handed to our controller i believe